scripts/pylint_extensions.py

```diff
@@ -794,6 +794,55 @@ def visit_functiondef(self, node):
             self.add_message('function-args-order-cls', node=node)
 
 
+class RestrictedImportChecker(checkers.BaseChecker):
+    """Custom pylint checker which checks restricted imports."""
+
+    __implements__ = interfaces.IAstroidChecker
+    name = 'invalid-import'
+    priority = -1
+    msgs = {
+        'C0009': (
+            'Importing domain layer in storage layer is prohibited.',
+            'invalid-import',
+            'Storage layer(core/storage) importing domain layer(core/domain)'
+            ' is not allowed.'),
+    }
+
+    def visit_import(self, node):
+        """Visits every import statement in the file.
+
+        Args:
+         node: astroid.node_classes.Import. Node for a import statement
+                 in the AST.
+        """
+
+        modnode = node.root()
+        names = [name for name, _ in node.names]
+        if 'oppia.core.storage' in modnode.name and not '_test' in modnode.name:
```

Comment:
Can you please list out the possible values of `modnode.name`? In what cases does it have `_test`?

Reply:
`modnode.name` is actually the name of the file being linted and as pointed out by @seanlip [here](https://github.com/oppia/oppia/pull/6180#discussion_r254433231) test files are to be excluded and `_test` is for such cases.
Thanks.